### PR TITLE
Reorganize codebase

### PR DESCRIPTION
**Pull request by ffaraoneim**
Load report, product, and customer command groups as plugins

### CODE REVIEW

File: docs/core_usage.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# CloudBlue Connect CLI core functionalities
+
+## Accounts management
+
+`connect-cli` need to configure at least one account to authenticate againts the
```
Comment (jpdolz) [inline docs/core_usage.md:5]: againts -> against

File: docs/customers_usage.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+$ ccli customer
+
+Usage: ccli customer [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  --help  Show this message and exit.
+
+Commands:
+  export  Export customers to an excel file.
+  sync    Syncronize customers from an excel file.
```
Comment (jpdolz) [inline docs/customers_usage.md:18]: Syncronize -> Synchronize

**Review by jpdolz — CHANGES_REQUESTED** [review]

File: connect/cli/core/constants.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+
+# This file is part of the Ingram Micro Cloud Blue Connect connect-cli.
+# Copyright (c) 2019-2021 Ingram Micro. All Rights Reserved.
+
+DEFAULT_ENDPOINT = 'https://api.connect.cloudblue.com/public/v1'
+DEFAULT_USER_AGENT = 'CloudBlue Connect CLI/'
+DEFAULT_BAR_FORMAT = '{desc:<70.69}{percentage:3.0f}%|{bar:30}{r_bar}'
+
+CAIRO_NOT_FOUND_ERROR = """Connect CLI depends on Cairo which appears not to be installed on the system.
```
[the remark this answers is not in the document]
Comment (marcserrat) [reply connect/cli/core/constants.py:10]: ...on Cairo which is not present on .....

[the remark this answers is not in the document]
Comment (ffaraoneim) [reply connect/cli/core/constants.py]: 👍

File: connect/cli/plugins/customer/commands.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        api_url=config.active.endpoint,
+        api_key=config.active.api_key,
+        output_file=output_file,
+        silent=config.silent,
+        output_path=output_path,
+        account_id=acc_id,
+    )
+    if not config.silent:
+        click.secho(
+            f'\nCustomers of account {acc_id} has been successfully exported to {outfile}',
```
[the remark this answers is not in the document]
Comment (ffaraoneim) [reply connect/cli/plugins/customer/commands.py]: 👍

File: connect/cli/plugins/customer/commands.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            fg='green',
+        )
+
+
+@grp_customer.command(
+    name='sync',
+    short_help='Syncronize customers from an excel file.',
```
[the remark this answers is not in the document]
Comment (ffaraoneim) [reply connect/cli/plugins/customer/commands.py]: 👍

File: connect/cli/plugins/product/commands.py [hunk 6]
```diff
@@ -138,13 +138,13 @@ def cmd_dump_products(config, product_id, output_file, output_path):
...
                 fg='green',
-            )
+            ),
         )
 
 
 @grp_product.command(
     name='sync',
-    short_help='sync a product from an excel file',
+    short_help='Syncronize a product from an excel file.',
```
[the remark this answers is not in the document]
Comment (ffaraoneim) [reply connect/cli/plugins/product/commands.py]: 👍

File: docs/core_usage.md (hunk 1 above)
Comment (ffaraoneim) [reply docs/core_usage.md]: 👍

File: docs/customers_usage.md (hunk 2 above)
Comment (ffaraoneim) [reply docs/customers_usage.md]: 👍

**Review by marcserrat — APPROVED** [review]

**Pull request merged**